### PR TITLE
replace an unhandled crash with a handled crash

### PR DESCRIPTION
**Pull request by sandlerr**
@zendesk/vegemite bring our handling of invalid json when polling the job status in line with the way we handle it everywhere else

### CODE REVIEW

**Comment by danielbreves** [comment]
👍

**Comment by ocke** [comment]
👍

File: lib/zendesk_apps_tools/deploy.rb [hunk 1]
```diff
@@ -8,19 +8,20 @@ def deploy_app(connection_method, url, body)
...
         req.headers[:content_type] = 'application/json'
-
         req.body = JSON.generate body
       end
 
       check_status response
 
     rescue Faraday::Error::ClientError => e
       say_error_and_exit e.message
+    rescue JSON::ParserError => e
```
Comment (svizzari) [inline lib/zendesk_apps_tools/deploy.rb]: I believe:
```ruby
rescue Faraday::Error::ClientError, JSON::ParserError => e
  say_error_and_exit e.message
end
```

would be more concise, but may invalidate the reason for this PR?

**Comment by sandlerr** [comment]
@svizzari refactored

**Comment by ocke** [comment]
👍

**Pull request merged**